sdk/core/azure-core/inc/azure/core/internal/strings.hpp

```diff
@@ -30,6 +30,22 @@ namespace Azure { namespace Core { namespace _internal {
       return (c < 'A' || c > 'Z') ? c : c + ('a' - 'A');
     }
 
+    static constexpr bool IsDigit(char c) noexcept { return c >= '0' && c <= '9'; }
+
+    static constexpr bool IsHexDigit(char c) noexcept
+    {
+      return IsDigit(c) || (c >= 'A' && c <= 'F') || (c >= 'a' && c <= 'f');
+    }
+
+    static constexpr bool IsAlphaNumeric(char c) noexcept
+    {
+      return IsDigit(c) || (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z');
+    }
+
+    static constexpr bool IsSpace(char c) noexcept { return c == ' ' || (c >= '\t' && c <= '\r'); }
```

Comment:
Do we really need this method, especially to support any space character other than`' '`?
The only caller I see is in user-agent string generation where we trim the whitespace from an `applicationId`.

I think we should move this implementation into `user_agent.cpp` to avoid increasing "publicly" exposed surface area, and only support trimming white space (unless we know we need to support the other space characters).

Reply:
As I said in another comment, the right way to evolve the code is to add `isxxx()` alternative here once it gets used anywhere in the code.

Reply:
It is an internal class, and I think that the right way to evolve that code is to keep adding `isxxxx()` to that class when `isxxxx()` is used at least once. Everything similar is in once place. It will be easier to maintain that long term. Otherwise you'll need to update 2 places in code when it gets used the second time (hopefully you can find other custom implementations with Ctrl+F), and also even when you add it the first time, you need a comment in code, why are we doing that at all. If we add it where it should be, everything is simple and is organized.